src/sparseml/optim/manager.py

```diff
@@ -333,33 +395,77 @@ def iter_modifiers(self) -> Generator[None, None, BaseModifier]:
             for mod in modifiers_list:
                 yield mod
 
-    def to_string_lines(self) -> List[str]:
+    def to_string_lines(self, include_metadata: bool = False) -> List[str]:
         """
+        :param include_metadata: boolean indicator whether metadata shall be
+            appended to the yaml file before saving. Default is False.
         :return: a list of lines for a string / yaml representation of this instance
         """
         yaml_str_lines = ["version: 1.1.0", ""]
+        # parse standard recipe
         if isinstance(self.modifiers, List):
+            if include_metadata and self._metadata:
+                yaml_str_lines.extend(self.metadata_to_string_lines())
             yaml_str_lines.append("modifiers:")
-        yaml_str_lines.extend(self.modifiers_to_string_lines(self.modifiers))
+            yaml_str_lines.extend(self.modifiers_list_to_string_lines(self.modifiers))
+        # parse staged recipe
+        else:
+            yaml_str_lines.extend(
+                self.modifiers_to_string_lines(self.modifiers, include_metadata)
+            )
+
+        return yaml_str_lines
+
+    def metadata_to_string_lines(self, stage: str = None) -> List[str]:
+        """
+        Parse `self._metadata` into list of strings.
+        :param stage: Name of the current recipe stage.
+            If stage = None, we are dealing with standard, unstaged recipe.
+        :return: a list of lines for a string / yaml representation of the
+            metadata for the given stage in the manager
+        """
+        yaml_str_lines = []
 
+        if stage:
+            yaml_str_lines.append(f"  {RECIPE_METADATA_KEY}:")
+            if not isinstance(self._metadata[stage], dict):
+                yaml_str_lines[-1] += f" {self._metadata[stage]}"
+            else:
+                for key, value in self._metadata[stage].items():
+                    yaml_str_lines.append(f"    {key}: {value}")
+        else:
+            yaml_str_lines.append(f"{RECIPE_METADATA_KEY}:")
+            if not isinstance(self._metadata[stage], dict):
```

Comment:
isn't `stage` False here? how should this work?

Reply:
This does not work obviously and should have been addressed before. Fixed now.